shared/reducers/fs.js

```diff
@@ -9,10 +9,18 @@ import * as Types from '../constants/types/fs'
 
 const initialState = Constants.makeState()
 
-const updatePathItem = (oldPathItem?: ?Types.PathItem, newPathItem: Types.PathItem): Types.PathItem => {
-  if (!oldPathItem || oldPathItem.type !== newPathItem.type) {
-    return newPathItem
+const updatePathItem = (
+  oldPathItem?: ?Types.PathItem,
+  newPathItemFromAction: Types.PathItem
+): Types.PathItem => {
+  if (!oldPathItem || oldPathItem.type !== newPathItemFromAction.type) {
+    return newPathItemFromAction
   }
+  // Reuse prefetchStatus if they equal in value.
+  // $FlowIssue
+  const newPathItem = newPathItemFromAction.update('prefetchStatus', newPrefetchStatus =>
+    newPrefetchStatus.equals(oldPathItem.prefetchStatus) ? oldPathItem.prefetchStatus : newPrefetchStatus
+  )
```

Comment:
What does this get us? If they're equal, then why not let it be set? You're calling `update` anyway so you get a new object no matter what...

Reply:
This re-uses the old `prefetchStatus` record if they equal in value. Update doesn't actually make a new immutable record unless we are giving it a different thing.

```
> r.a
123
> r.set('a', 123) === r
true
```

Reply:
OK but if you're using the same references everywhere for a given status, then it shouldn't matter, right?

Reply:
For enabled/disabled, you are right it wouldn't matter. But for partially enabled ones, it's not the same reference everywhere. So I thought I might as well do it right here in the first place.

Reply:
OK fair enough, thanks for explaining! Mind putting a comment?